sdk/search/azure-search-documents/swagger/src/main/java/SearchIndexCustomizations.java

```diff
@@ -1,11 +1,88 @@
+        classCustomization.getMethod("setOrderBy").replaceParameters("String... orderBy")
+            .replaceBody(String.format(VARARG_METHOD_TEMPLATE, "orderBy", "orderBy", "orderBy"));
+        classCustomization.getMethod("setSearchFields").replaceParameters("String... searchFields")
+            .replaceBody(String.format(VARARG_METHOD_TEMPLATE, "searchFields", "searchFields", "searchFields"));
+        classCustomization.getMethod("setSelect").replaceParameters("String... select")
+            .replaceBody(String.format(VARARG_METHOD_TEMPLATE, "select", "select", "select"));
+    }
+
+    private void customizeImplementationModelsPackage(PackageCustomization packageCustomization) {
+        customizeSearchOptions(packageCustomization.getClass(SEARCH_OPTIONS));
+    }
+
+    private void customizeSearchOptions(ClassCustomization classCustomization) {
+        classCustomization.getMethod("isIncludeTotalCount").rename("isTotalCountIncluded");
+
+        classCustomization.getMethod("setFacets").replaceParameters("String... facets")
+            .replaceBody(String.format(VARARG_METHOD_TEMPLATE, "facets", "facets", "facets"));
+
+        classCustomization.getMethod("setOrderBy").replaceParameters("String... orderBy")
+            .replaceBody(String.format(VARARG_METHOD_TEMPLATE, "orderBy", "orderBy", "orderBy"));
+
+        classCustomization.getMethod("setSearchFields").replaceParameters("String... searchFields")
+            .replaceBody(String.format(VARARG_METHOD_TEMPLATE, "searchFields", "searchFields", "searchFields"));
+
+        classCustomization.getMethod("setSelect").replaceParameters("String... select")
+            .replaceBody(String.format(VARARG_METHOD_TEMPLATE, "select", "select", "select"));
+
+        classCustomization.getMethod("setHighlightFields").replaceParameters("String... highlightFields")
+            .replaceBody(String.format(VARARG_METHOD_TEMPLATE, "highlightFields", "highlightFields", "highlightFields"));
+
+        // Can't be done right now as setScoringParameters uses String.
+//        // Scoring parameters are slightly different as code generates them as String.
+//        classCustomization.getMethod("setScoringParameters").replaceParameters("ScoringParameter... scoringParameters")
+//            .replaceBody(
+//                "this.scoringParameters = (scoringParameters == null)"
+//                    + "    ? null"
+//                    + "    : java.util.Arrays.stream(scoringParameters)"
+//                    + "        .map(ScoringParameter::toString)"
+//                    + "        .collect(java.util.stream.Collectors.toList());"
+//                    + "return this;");
+//
+//        classCustomization.getMethod("getScoringParameters")
+//            .setReturnType("List<ScoringParameter>",
+//                "this.scoringParameters.stream().map(ScoringParameter::new).collect(java.util.stream.Collectors.toList())");
```

Comment:
Can this be removed and added later when this is supported?

Reply:
I'd rather leave it as a reminder of what needs to change to enable the code generated class matching what the hand-written class provides